datalad/distribution/add.py

```diff
@@ -404,7 +437,10 @@ def __call__(
                     # in the dataset
                     ap = {k: v for k, v in res.items() if k != 'status'}
                     ap['staged'] = True
-                    to_save.append({k: v for k, v in res.items() if k != 'status'})
+                    # strip any status and state info (e.g. save will refuse to save
+                    # stuff that is marked state='untracked'
```

Comment:
I agree. To me it's just about consistency. It might be hard to find that spot, if that unknown state later turns out to be troublesome.